Adding jgit Dependency for REST API

PhilMFischer opened the pull request:
Based on PR #362 .
Closes #359.

- Added jgit dependencies to project
- Started initial implementation for rest service on git persistance layer
- Got initial setup running in osgi environment
- Got test cases running

- [x] test cases for git service need to be improved

Commit:
Added initial projects - (Task #356)

- Added all relevant projects and features for the new server
- Added plugin project
- Added feature
- Added test fragment
- Added tests to all tests
- Adjusted POMS


---
Task #356: Jetty baseline server based on Virtual Satellite APPs

Commit:
Merge branch 'feature/task_355-Add_Jersey_and_Jetty_' into feature/task_356-Jetty_baseline_server

Commit:
Added first simple implementation for Jersey rest API - (Task #356)

- Added mock object for workspace level access
- Added mock object for model level access

---
Task #356: Jetty baseline server based on Virtual Satellite APPs

Commit:
Renaming because of confused eclipse - (Task #356)

- eclispe and git gets confused with the ignore settings for "workspace"
directories.

---
Task #356: Jetty baseline server based on Virtual Satellite APPs

dianapeters-dlr committed:
Add dependency for jgit - (Task #359)

... and start example implementation (not yet running)

---
Task #359: Add JGit dependencies and make it available via rest
interface

dianapeters-dlr committed:
Add jgit dependencies - (Task #359)

Now really (forgot files in last commit).

---
Task #359: Add JGit dependencies and make it available via rest
interface

Commit:
Added signatures for Commit and Update - (Task #359)

- Added commit signature
- Added update signature

---
Task #359: Add JGit dependencies and make it available via rest
interface

Commit:
Fixing host fragment for server tests - (Task #356)

- Server tests were still bound to team. This is fixed now.

---
Task #356: Jetty baseline server based on Virtual Satellite APPs

Commit:
Merge branch 'feature/task_356-Jetty_baseline_server' into feature/task_359-Add_JGit_dependencies

Commit:
Starting to add tests cases for Jetty server - (Task #359)

- Added first abstract test
- Changing code structure of Server itself, to start an instance from
the test case.

---
Task #359: Add JGit dependencies and make it available via rest
interface

dianapeters-dlr committed:
First attempt for clone - (Task #359)

Clone of a repository via JGit works in principle, including
credentials.

---
Task #359: Add JGit dependencies and make it available via rest
interface

dianapeters-dlr committed:
Merge branch 'feature/task_359-Add_JGit_dependencies' of https://github.com/virtualsatellite/VirtualSatellite4-Core into feature/task_359-Add_JGit_dependencies

Commit:
Started to implement test cases - (Task #359)

- Trying to setup git repository beforehand.

---
Task #359: Add JGit dependencies and make it available via rest
interface

dianapeters-dlr committed:
Start impl. of commit feature w. JGit - (Task #359)

First try of usage of the the commit feature of JGit (add all files,
commit, push).

---
Task #359: Add JGit dependencies and make it available via rest
interface

dianapeters-dlr committed:
Merge branch 'feature/task_359-Add_JGit_dependencies' of https://github.com/virtualsatellite/VirtualSatellite4-Core into feature/task_359-Add_JGit_dependencies

Commit:
Fixing bundle names in manifest - (Task #359)

- Corrected names for automatic java modules

---
Task #359: Add JGit dependencies and make it available via rest
interface

Commit:
Fixing dependencies in features - (Task #359)

- Fixing dependencies in features for the jetty server and for all tests
- Also fixed dependency in feature for new external libs. Jersey was
missing.

---
Task #359: Add JGit dependencies and make it available via rest
interface

Commit:
Fixed launch config for all tests - (Task #359)

- All tests is now using the new server feature as well.

---
Task #359: Add JGit dependencies and make it available via rest
interface

Commit:
Merged with changes of diana - (Task #359)

- Merged incoming changes.

---
Task #359: Add JGit dependencies and make it available via rest
interface

Commit:
Changed import of jgit in Manifest file - (Task #359)

- changed from indirect package import to direct plugin import of jgit
dependency

---
Task #359: Add JGit dependencies and make it available via rest
interface

dianapeters-dlr committed:
First impl. of pull with JGit - (Task #359)

Pull of repository changes with JGit works. Needs cleanup.

---
Task #359: Add JGit dependencies and make it available via rest
interface

Commit:
Adding commons IO as dependencies - (Task #359)

- Added commons io and source from orbit repositories to target platform

---
Task #359: Add JGit dependencies and make it available via rest
interface

Commit:
Merge branch 'feature/task_359-Add_JGit_dependencies' of https://github.com/virtualsatellite/VirtualSatellite4-Core.git into feature/task_359-Add_JGit_dependencies

dianapeters-dlr committed:
Fix extraction of repo directory name - (Task #359)

Use apache.commons.io to extract directory name from git URI.

---
Task #359: Add JGit dependencies and make it available via rest
interface

dianapeters-dlr committed:
Merge branch 'feature/task_359-Add_JGit_dependencies' of https://github.com/virtualsatellite/VirtualSatellite4-Core into feature/task_359-Add_JGit_dependencies

dianapeters-dlr committed:
Remove username+password for git access - (Task #359)

Removed support for credentials - only local repositories are supported
now.

---
Task #359: Add JGit dependencies and make it available via rest
interface

dianapeters-dlr committed:
Add param for local directory - (Task #359)

Add queryparam to all endpoints which allows user to specify where the
repository lies locally (or should be cloned to). Discard URI at some
points, because the local directory is sufficient (for commit and
update).

---
Task #359: Add JGit dependencies and make it available via rest
interface

Commit:
Fixed issue with ServletContainer in osgi environment - (Task #359)

- Added javax.servelet as dependency to external.lib.jersey. This makes
the classloader aware of the ServeletContainer from glasfish.jetty is
compatible to javax.Servelet.
- Removed dependencies to jersey from target-platform. used to be taken
from orbit repos.
- Refactored Server to be startable and stopable as server from unit
tests
- Added generic unit test that can be sued for verifying all rest
implementations

---
Task #359: Add JGit dependencies and make it available via rest
interface

Commit:
Merge branch 'feature/task_359-Add_JGit_dependencies' of https://github.com/virtualsatellite/VirtualSatellite4-Core.git into feature/task_359-Add_JGit_dependencies

Commit:
removed version infos from dependencies - (Task #359)

- Removed version infos from server plugin.

---
Task #359: Add JGit dependencies and make it available via rest
interface

Commit:
Fixed dependency in external lib feature - (Task #359)

- Removed old jersey dependency

---
Task #359: Add JGit dependencies and make it available via rest
interface

Commit:
Added Diana to known Authors - (Task #359)

- updated mailmap file
- updated known authors file

---
Task #359: Add JGit dependencies and make it available via rest
interface

Commit:
Merging with jetty which has been added as dependency - (Task #359)

- Added jetty dependency
- Merged manifest and feature.xml for new dependency

---
Task #359: Add JGit dependencies and make it available via rest
interface

dianapeters-dlr committed:
Rename MAR to ATR - (Task #359)

Rename the resource which is currently used for testing.

---
Task #359: Add JGit dependencies and make it available via rest
interface

Commit:
Added sources to jersey plugins for betetr debugging - (Task #359)

- Added zip with jersey source code from the original github repo
- Added zip to classpath entries of external libs/jars

---
Task #359: Add JGit dependencies and make it available via rest
interface

Commit:
Introduced new Servlet for Equinox Environment - (Task #359)

- Added new Servlet thatd irectly registeres the Classes rather than
looking them up.
- Updated Test cases to make use of them

---
Task #359: Add JGit dependencies and make it available via rest
interface

Commit:
Merged and adjusted tests to use test resource servelet - (Task #359)

- Had to change uri from mar to atr context.

---
Task #359: Add JGit dependencies and make it available via rest
interface

Commit:
Merged with dependencies - (Task #355)



---
Task #355: Add Jersey and Jetty dependencies to project

dianapeters-dlr commented on de.dlr.sc.virsat.server.test/src/de/dlr/sc/virsat/server/resources/WorkspaceAccessResourceTest.java:
```diff
@@ -0,0 +1,86 @@
+//	@Test
+//	public void testUpdate() {
+//		fail("Not yet implemented");
+//	}
```
Remove.

PhilMFischer replied:
Fixed with further test case implementations

dianapeters-dlr commented on de.dlr.sc.virsat.server.test/src/de/dlr/sc/virsat/server/resources/WorkspaceAccessResourceTest.java:
```diff
@@ -0,0 +1,86 @@
+	}
+
+	@Test
+	public void testCloneString() {
```
Since it is known that we have todos for this class, I don't add more comment ;)

dianapeters-dlr commented on de.dlr.sc.virsat.server.test/src/de/dlr/sc/virsat/server/test/AllTests.java:
```diff
@@ -0,0 +1,42 @@
+ * 
+ * @author scha_vo
+ *
+ */
```
?

PhilMFischer replied:
Copy & paste from the past 😄

dianapeters-dlr commented on de.dlr.sc.virsat.server/about.html:
```diff
@@ -0,0 +1,27 @@
+<body lang="EN-US">
+<h2>About This Content</h2>
+ 
+<p>December 18, 2018</p>	
```
Same as above - change date?

dianapeters-dlr commented on de.dlr.sc.virsat.server/src/de/dlr/sc/virsat/server/VirSatJettyServer.java:
```diff
@@ -0,0 +1,77 @@
+	public VirSatJettyServer() {
+	}
+	
+	private static final int VIRSAT_JETTY_PORT = 8000; 
```
We wanted to change this back to 8080.

PhilMFischer replied:
I think i will open a ticket , to let the user hand over a command line argument to it.

dianapeters-dlr commented on de.dlr.sc.virsat.server/src/de/dlr/sc/virsat/server/VirSatJettyServer.java:
```diff
@@ -0,0 +1,77 @@
+	
+	/**
+	 *  Call this method to start the Virtual Satellite specific Jetty Server.
+	 *  This method blocks in a join statement to wait make the cally thread wait
```
I think the first "want" is too much - the sentence sounds gramatically wrong. And "cally"?

PhilMFischer replied:
True... fixed...

dianapeters-dlr commented on de.dlr.sc.virsat.server/src/de/dlr/sc/virsat/server/VirSatJettyServer.java:
```diff
@@ -0,0 +1,77 @@
+	}
+	
+	public VirSatJettyServer join() throws InterruptedException {
+		server.join();
```
Since there is a null-check in the stop method, it would make sense to have one here as well, doesn't it? (You can call join before start, after all.)

PhilMFischer replied:
it probably doesn'tharm… added.

dianapeters-dlr commented on de.dlr.sc.virsat.server/src/de/dlr/sc/virsat/server/VirSatModelAccessServlet.java:
```diff
@@ -0,0 +1,35 @@
+		return new ServletContainer(resourceConfig);
+	}
+
+	// @ApplicationPath(CONTEXT_PATH)
```
remove comment? (I don't see why it should remain here.)

PhilMFischer replied:
done

dianapeters-dlr commented on de.dlr.sc.virsat.server/src/de/dlr/sc/virsat/server/data/GitAccess.java:
```diff
@@ -0,0 +1,155 @@
+		return result;
+	}
+
+	/**
```
Add javadoc (not just the skeleton).

PhilMFischer replied:
Done...

dianapeters-dlr commented on de.dlr.sc.virsat.server/src/de/dlr/sc/virsat/server/data/GitAccess.java:
```diff
@@ -0,0 +1,155 @@
+		try {
+			git = Git.open(new File(localDirectory));
+			git.pull()
+				.call();
```
fix indentation

PhilMFischer replied:
no, it is the same pattern as for all the other calls. In this case it should be ok.

Commit:
Added Unit tests for new GitAccess class - (Task #359)

- Added unit tests

---
Task #359: Add JGit dependencies and make it available via rest
interface

Commit:
Merge remote-tracking branch 'origin/development' into feature/task_359-Add_JGit_dependencies

Commit:
AddedTests for ViorSat WorkspaceAccessResource - (Task #359)

- Added test Cases to check the REST API for repository calls
- simplified code
- Added static variables for REST path elements

---
Task #359: Add JGit dependencies and make it available via rest
interface

PhilMFischer commented on de.dlr.sc.virsat.external.lib.jersey/src/de/dlr/virsat/external/lib/jersey/servlet/ApplicationServletContainer.java:
```diff
@@ -0,0 +1,64 @@
+
+
+
+public abstract class ApplicationServletContainer implements Servlet {
```
🔨 fixed

PhilMFischer commented on de.dlr.sc.virsat.external.lib.jersey/src/de/dlr/virsat/external/lib/jersey/servlet/ApplicationServletContainer.java:
```diff
@@ -0,0 +1,64 @@
+
+	@Override
+	public String getServletInfo() {
+		return getServletInfo();
```
:hammer: fixed

PhilMFischer commented on de.dlr.sc.virsat.server.test/about.html:
```diff
@@ -0,0 +1,27 @@
+<body lang="EN-US">
+<h2>About This Content</h2>
+ 
+<p>December 18, 2018</p>	
```
I am also not sure, i leave it like this for the moment.

PhilMFischer commented on de.dlr.sc.virsat.server.feature/feature.xml:
```diff
@@ -0,0 +1,43 @@
+   <copyright>
+      Copyright 2015
+
+by German Aerospace Center (DLR e.V.)
```
Eclipse is doing that. Not in our hands.

Commit:
Changes due to Review - (Task #359)

- Adjsuted code to review comments

---
Task #359: Add JGit dependencies and make it available via rest
interface

Commit:
Removed redundant file seperator in test - (Task #359)

- Assert was expecting to many file seperators

---
Task #359: Add JGit dependencies and make it available via rest
interface

Commit:
Fixed call to testuite - (Task #356)

- Added tests to test suite.
- Improved jetty server shutdown in tests.


---
Task #356: Jetty baseline server based on Virtual Satellite APPs

SaMuellerDLR commented on de.dlr.sc.virsat.server.feature/feature.xml:
```diff
@@ -0,0 +1,43 @@
+      license-feature="de.dlr.sc.virsat.license.feature">
+
+   <description>
+      VirSat Team Connector
```
Probably should be VirSat Server or something. Looks like copy paste.

PhilMFischer replied:
I think this is fixed already in a different branch. I will check it.

SaMuellerDLR commented on de.dlr.sc.virsat.server.test/src/de/dlr/sc/virsat/server/data/VirSatGitAccessTest.java:
```diff
@@ -0,0 +1,157 @@
+
+public class VirSatGitAccessTest {
+
+	private static File pathToTempUpstreamRepository;
```
Any reason they are static? Data seems to be created for every test run.
Dont see why it needs to be persisted.

PhilMFischer replied:
No, i will remove the statics.

SaMuellerDLR commented on de.dlr.sc.virsat.server.test/src/de/dlr/sc/virsat/server/resources/AccessTestResourceTest.java:
```diff
@@ -0,0 +1,73 @@
+public class AccessTestResourceTest extends AGitAndJettyServerTest {
+
+	@Before
+	public void setUp() throws Exception {
```
No need to overwrite setUp/tearDown just to call their super methods.

PhilMFischer replied:
right.

SaMuellerDLR commented on de.dlr.sc.virsat.server.test/src/de/dlr/sc/virsat/server/resources/WorkspaceAccessResourceTest.java:
```diff
@@ -0,0 +1,131 @@
+		// Add a file to repo one and commit it
+		// now add a file to the local repository
+		File newFile = new File(pathToTempLocalRepository1.getAbsolutePath().toString() + "/test.dat");
+		Files.createFile(newFile.toPath());
```
Is there any reason why newFile.createNewFile(); doesnt work?
I think it would be a little bit cleaner.

PhilMFischer replied:
Will give it a try

SaMuellerDLR commented on de.dlr.sc.virsat.server/src/de/dlr/sc/virsat/server/data/VirSatGitAccess.java:
```diff
@@ -0,0 +1,110 @@
+	 */
+	public String update(String localDirectory) {
+		String result = STATUS_OK;
+		Git git;
```
I think the git variable can be moved into the try { } block.

Commit:
Fixes due to review - (Task #359)


---
Task #359: Add JGit dependencies and make it available via rest
interface

SaMuellerDLR reviewed (approved):
Approved!
